### PR TITLE
fix: refactor file upload

### PR DESCRIPTION
**Pull request by marybrown123**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -24,15 +24,25 @@ model Recipe {
...
+  fileId Int? @unique @map("file_id")
 
   @@map("recipe")
 }
 
+model File {
+  id Int @default(autoincrement()) @id
+  name String
+  key String
+  recipe Recipe?
```
Comment (wojdanowski) [inline prisma/schema.prisma:41]: I don't think referencing the recipe here is necessary. See prisma documentation. It's possible to define one to one relation on one side (in this case it should be on recipe side)
https://www.prisma.io/docs/concepts/components/prisma-schema/relations/one-to-one-relations

File: src/file/commands/handlers/delete-file.handler.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { DeleteFileCommand } from '../../commands/impl/delete-file.command';
+import { FileDAO } from '../../file.dao';
+import { FileResponse } from '../../responses/file.response';
+
+@CommandHandler(DeleteFileCommand)
+export class DeleteFileHandler implements ICommandHandler<DeleteFileCommand> {
+  constructor(private fileDAO: FileDAO) {}
+  async execute(command: DeleteFileCommand): Promise<FileResponse> {
+    const { fileId } = command;
+    return this.fileDAO.deleteFile(fileId);
```
Comment (wojdanowski) [inline src/file/commands/handlers/delete-file.handler.ts:11]: when the file is removed from the database you should also remove it from s3

File: src/file/file.controller.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+} from '@nestjs/swagger';
+import { FileService } from '../file/file.service';
+import { FileResponse } from '../file/responses/file.response';
+import { S3Service } from '../file/s3.service';
+
+@Controller('/file')
+export class FileController {
+  constructor(
+    private readonly fileService: FileService,
+    private readonly s3Service: S3Service,
```
Comment (wojdanowski) [inline src/file/file.controller.ts:23]: you do not need it here

File: src/file/file.controller.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  @Post()
+  @UseInterceptors(FileInterceptor('file'))
+  @UseGuards(AuthGuard('jwt'))
+  @ApiOperation({ summary: 'Create a new file' })
+  @ApiCreatedResponse({ type: FileResponse })
+  @ApiUnauthorizedResponse({ description: 'Not logged in' })
+  async createFile(
+    @UploadedFile() file: Express.Multer.File,
+  ): Promise<FileResponse> {
+    const fileKey = await this.s3Service.uploadFile(file);
```
Comment (wojdanowski) [inline src/file/file.controller.ts:35]: I think that command handler would be a good place to place this logic. 
That way you would have a clean controller and all the "business" logic in one place.

File: src/recipe/recipe.dao.ts [hunk 5]
```diff
@@ -1,20 +1,32 @@
...
+import { FileService } from '../file/file.service';
+import { S3Service } from '../file/s3.service';
 
 @Injectable()
 export class RecipeDAO {
-  constructor(private prismaService: PrismaService) {}
+  constructor(
+    private prismaService: PrismaService,
+    private fileService: FileService,
+    private s3Service: S3Service,
```
Comment (wojdanowski) [inline src/recipe/recipe.dao.ts:15]: consider moving those (fileService, s3Service) outside the recipeDao

**Review by wojdanowski — APPROVED** [review]

File: src/file/commands/handlers/delete-file.handler.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import { S3Service } from '../../s3.service';
+
+@CommandHandler(DeleteFileCommand)
+export class DeleteFileHandler implements ICommandHandler<DeleteFileCommand> {
+  constructor(private fileDAO: FileDAO, private s3Service: S3Service) {}
+  async execute(command: DeleteFileCommand): Promise<void> {
+    const { fileId } = command;
+    const fileToDelete = await this.fileDAO.findFileById(fileId);
+    await this.s3Service.deleteFile(fileToDelete.key);
+    this.fileDAO.deleteFile(fileId);
```
Comment (wojdanowski) [inline src/file/commands/handlers/delete-file.handler.ts:13]: no await?

Comment (marybrown123) [reply src/file/commands/handlers/delete-file.handler.ts:13]: my mistake

**Pull request merged**